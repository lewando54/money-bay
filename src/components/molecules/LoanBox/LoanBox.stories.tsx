/* eslint-disable @typescript-eslint/consistent-type-assertions */
import { StoryObj, Meta } from '@storybook/react-native'

import LoanBox from './LoanBox'

const meta = {
    component: LoanBox,
    argTypes: {
        amount: { control: 'number' },
        rate: { control: 'number' },
        period: { control: 'number' },
        monthlyPayment: { control: 'number' },
        totalPaid: { control: 'number' },
        currency: { control: 'object' },
        onPress: { action: 'clicked' }
    }
} satisfies Meta<typeof LoanBox>

export default meta
type Story = StoryObj<typeof meta>

export const Default: Story = {
    args: {
        amount: -20532,
        rate: 13,
        period: 24,
        monthlyPayment: 1117,
        totalPaid: 4468,
        currency: {iso: 'usd', symbol: '$', left: true},
    }
}

export const OneMonth: Story = {
    args: {
        amount: -20532,
        rate: 13,
        period: 1,
        monthlyPayment: 1117,
        totalPaid: 4468,
        currency: {iso: 'usd', symbol: '$', left: true},
    }
}
