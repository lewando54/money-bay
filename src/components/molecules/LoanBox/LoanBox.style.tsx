import { StyleSheet } from 'react-native'
export default StyleSheet.create({
    container: {
        gap: 20
    },
    header: {
        flexDirection: 'row',
        alignItems: 'center',
        gap: 10
    },
    repayBtn: {
        marginLeft: 'auto'
    },
    innerContainer: {
        flexWrap: 'wrap',
        gap: 16
    },
    innerRow: {
        flexDirection: 'row'
    },
    innerRowItem: {
        width: '50%'
    }
})
