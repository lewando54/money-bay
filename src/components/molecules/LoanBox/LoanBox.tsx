import React from 'react'
import { View, Text } from 'react-native'
import Separator from '../../atoms/Separator/Separator'
import UniversalContainer from '../../atoms/UniversalContainer/UniversalContainer'
import Wallet from './assets/wallet.svg'
import MoneyText from '../../atoms/MoneyText/MoneyText'
import Button from '../../atoms/Button/Button'
import globalThemeStyle, { GOOD_GREEN } from '../../../styling/GlobalTheme.style'
import { TCurrency } from '../../../utils/currency'

interface ILoanBoxProps {
    amount?: number
    rate?: number
    period?: number
    monthlyPayment?: number
    totalPaid?: number
    currency?: TCurrency
    onPress?: () => void
}

export default function LoanBox({
    amount=-20532,
    rate=13,
    period=24,
    monthlyPayment=1117,
    totalPaid=4468,
    currency={iso: 'usd', symbol: '$', left: true},
    onPress
}: ILoanBoxProps){
    return (
        <UniversalContainer padding={20} style={{gap: 20}} castShadow={true}>
            <View style={[{flexDirection: 'row', alignItems: 'center', gap: 10}]}>
                <Wallet />
                <MoneyText amount={amount} hasNumberFormatting={true}/>
                <Button size='small' width={'30%'} color={GOOD_GREEN} containerStyle={{marginLeft: 'auto'}} onClick={onPress}>Repay</Button>
            </View>
            <Separator />
            <View style={{flexWrap: 'wrap', gap: 16}}>
                <View style={{flexDirection: 'row'}}>
                    <View style={{width: '50%'}}>
                        <Text style={globalThemeStyle.text_SubTitle}>Rate</Text>
                        <Text style={globalThemeStyle.text_Title}>{rate}</Text>
                    </View>
                    <View style={{width: '50%'}}>
                        <Text style={globalThemeStyle.text_SubTitle}>Period</Text>
                        <Text style={globalThemeStyle.text_Title}>{period} month{period == 1 ? '' : 's'}</Text>
                    </View>
                </View>
                <View style={{flexDirection: 'row'}}>
                    <View style={{width: '50%'}}>
                        <Text style={globalThemeStyle.text_SubTitle}>Monthly payment</Text>
                        <Text style={globalThemeStyle.text_Title}>{monthlyPayment.toLocaleString(undefined, {minimumFractionDigits: 2, minimumIntegerDigits: 2}).replace(',', ' ')+' '+currency.iso.toLocaleUpperCase()}</Text>
                    </View>
                    <View style={{width: '50%'}}>
                        <Text style={globalThemeStyle.text_SubTitle}>Total paid</Text>
                        <Text style={globalThemeStyle.text_Title}>{totalPaid.toLocaleString(undefined, {minimumFractionDigits: 2, minimumIntegerDigits: 2}).replace(',', ' ')+' '+currency.iso.toLocaleUpperCase()}</Text>
                    </View>
                </View>
            </View>
        </UniversalContainer>
    )
}
