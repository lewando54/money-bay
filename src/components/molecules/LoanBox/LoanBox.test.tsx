import React from 'react'
import renderer from 'react-test-renderer'
import LoanBox from './LoanBox'
describe('LoanBox', () => {
    it('should render correctly', () => {
        const tree = renderer.create(<LoanBox />)
        expect(tree).toMatchSnapshot()
    })

    it('should render one month correctly', () => {
        const tree = renderer.create(<LoanBox period={1}/>)
        expect(tree).toMatchSnapshot()
    })
})
