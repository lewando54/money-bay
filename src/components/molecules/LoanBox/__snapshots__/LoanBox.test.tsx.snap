// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoanBox should render correctly 1`] = `
<View
  style={
    [
      {
        "backgroundColor": "white",
        "borderRadius": 10,
        "height": "auto",
        "padding": 20,
        "width": "100%",
      },
      {
        "elevation": 20,
        "shadowColor": "#16196670",
        "shadowOffset": {
          "height": 6,
          "width": 0,
        },
        "shadowOpacity": 0.08,
        "shadowRadius": 16,
      },
      {
        "gap": 20,
      },
    ]
  }
  testID="universal-container"
>
  <View
    style={
      {
        "alignItems": "center",
        "flexDirection": "row",
        "gap": 10,
      }
    }
  >
    <SvgMock />
    <View
      style={
        [
          {
            "alignItems": "flex-end",
            "flexDirection": "row",
          },
          undefined,
        ]
      }
    >
      <Text
        style={
          [
            {
              "color": "white",
              "fontFamily": "MulishRegular",
            },
            {
              "color": "#1B1D4D",
              "fontSize": 20,
            },
          ]
        }
      >
        - 
      </Text>
      <Text
        style={
          [
            {
              "color": "white",
              "fontFamily": "MulishRegular",
            },
            {
              "color": "#1B1D4D",
              "fontSize": 20,
            },
          ]
        }
      >
        20 532
        .
      </Text>
      <View
        style={
          [
            {
              "alignItems": "flex-end",
              "flexDirection": "row",
            },
            {
              "paddingBottom": 2,
            },
          ]
        }
      >
        <Text
          style={
            [
              {
                "color": "white",
                "fontFamily": "MulishRegular",
              },
              {
                "color": "#1B1D4D",
                "fontSize": 14,
              },
            ]
          }
        >
          00
        </Text>
        <Text
          style={
            [
              {
                "color": "white",
                "fontFamily": "MulishBold",
              },
              {
                "color": "#1B1D4D",
                "fontSize": 14,
                "marginLeft": 4,
              },
            ]
          }
        >
          USD
        </Text>
      </View>
    </View>
    <View
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": undefined,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={false}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "marginLeft": "auto",
          "opacity": 1,
          "width": "30%",
        }
      }
    >
      <ViewManagerAdapter_ExpoLinearGradient
        proxiedProperties={
          {
            "colors": [
              4282299024,
              4282299024,
            ],
            "endPoint": [
              1,
              0.5,
            ],
            "locations": undefined,
            "startPoint": [
              0,
              0.5,
            ],
          }
        }
        style={
          [
            {
              "borderRadius": 10,
              "padding": 15,
            },
            {
              "borderRadius": 5,
              "padding": 8,
            },
            false,
          ]
        }
      >
        <Text
          style={
            [
              {
                "fontFamily": "MulishSemiBold",
                "fontSize": 16,
                "textAlign": "center",
              },
              false,
              false,
              [
                {
                  "fontSize": 12,
                },
                {
                  "color": "white",
                  "fontFamily": "MulishSemiBold",
                },
              ],
            ]
          }
        >
          Repay
        </Text>
      </ViewManagerAdapter_ExpoLinearGradient>
    </View>
  </View>
  <View
    style={
      [
        {
          "backgroundColor": "#CED6E1",
          "height": 1,
          "width": "100%",
        },
        undefined,
      ]
    }
  />
  <View
    style={
      {
        "flexWrap": "wrap",
        "gap": 16,
      }
    }
  >
    <View
      style={
        {
          "flexDirection": "row",
        }
      }
    >
      <View
        style={
          {
            "width": "50%",
          }
        }
      >
        <Text
          style={
            {
              "color": "#4C4C60",
              "fontFamily": "MulishRegular",
              "fontSize": 12,
            }
          }
        >
          Rate
        </Text>
        <Text
          style={
            {
              "color": "#1B1D4D",
              "fontFamily": "MulishSemiBold",
              "fontSize": 14,
            }
          }
        >
          13
        </Text>
      </View>
      <View
        style={
          {
            "width": "50%",
          }
        }
      >
        <Text
          style={
            {
              "color": "#4C4C60",
              "fontFamily": "MulishRegular",
              "fontSize": 12,
            }
          }
        >
          Period
        </Text>
        <Text
          style={
            {
              "color": "#1B1D4D",
              "fontFamily": "MulishSemiBold",
              "fontSize": 14,
            }
          }
        >
          24
           month
          s
        </Text>
      </View>
    </View>
    <View
      style={
        {
          "flexDirection": "row",
        }
      }
    >
      <View
        style={
          {
            "width": "50%",
          }
        }
      >
        <Text
          style={
            {
              "color": "#4C4C60",
              "fontFamily": "MulishRegular",
              "fontSize": 12,
            }
          }
        >
          Monthly payment
        </Text>
        <Text
          style={
            {
              "color": "#1B1D4D",
              "fontFamily": "MulishSemiBold",
              "fontSize": 14,
            }
          }
        >
          1 117.00 USD
        </Text>
      </View>
      <View
        style={
          {
            "width": "50%",
          }
        }
      >
        <Text
          style={
            {
              "color": "#4C4C60",
              "fontFamily": "MulishRegular",
              "fontSize": 12,
            }
          }
        >
          Total paid
        </Text>
        <Text
          style={
            {
              "color": "#1B1D4D",
              "fontFamily": "MulishSemiBold",
              "fontSize": 14,
            }
          }
        >
          4 468.00 USD
        </Text>
      </View>
    </View>
  </View>
</View>
`;

exports[`LoanBox should render one month correctly 1`] = `
<View
  style={
    [
      {
        "backgroundColor": "white",
        "borderRadius": 10,
        "height": "auto",
        "padding": 20,
        "width": "100%",
      },
      {
        "elevation": 20,
        "shadowColor": "#16196670",
        "shadowOffset": {
          "height": 6,
          "width": 0,
        },
        "shadowOpacity": 0.08,
        "shadowRadius": 16,
      },
      {
        "gap": 20,
      },
    ]
  }
  testID="universal-container"
>
  <View
    style={
      {
        "alignItems": "center",
        "flexDirection": "row",
        "gap": 10,
      }
    }
  >
    <SvgMock />
    <View
      style={
        [
          {
            "alignItems": "flex-end",
            "flexDirection": "row",
          },
          undefined,
        ]
      }
    >
      <Text
        style={
          [
            {
              "color": "white",
              "fontFamily": "MulishRegular",
            },
            {
              "color": "#1B1D4D",
              "fontSize": 20,
            },
          ]
        }
      >
        - 
      </Text>
      <Text
        style={
          [
            {
              "color": "white",
              "fontFamily": "MulishRegular",
            },
            {
              "color": "#1B1D4D",
              "fontSize": 20,
            },
          ]
        }
      >
        20 532
        .
      </Text>
      <View
        style={
          [
            {
              "alignItems": "flex-end",
              "flexDirection": "row",
            },
            {
              "paddingBottom": 2,
            },
          ]
        }
      >
        <Text
          style={
            [
              {
                "color": "white",
                "fontFamily": "MulishRegular",
              },
              {
                "color": "#1B1D4D",
                "fontSize": 14,
              },
            ]
          }
        >
          00
        </Text>
        <Text
          style={
            [
              {
                "color": "white",
                "fontFamily": "MulishBold",
              },
              {
                "color": "#1B1D4D",
                "fontSize": 14,
                "marginLeft": 4,
              },
            ]
          }
        >
          USD
        </Text>
      </View>
    </View>
    <View
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": undefined,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={false}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "marginLeft": "auto",
          "opacity": 1,
          "width": "30%",
        }
      }
    >
      <ViewManagerAdapter_ExpoLinearGradient
        proxiedProperties={
          {
            "colors": [
              4282299024,
              4282299024,
            ],
            "endPoint": [
              1,
              0.5,
            ],
            "locations": undefined,
            "startPoint": [
              0,
              0.5,
            ],
          }
        }
        style={
          [
            {
              "borderRadius": 10,
              "padding": 15,
            },
            {
              "borderRadius": 5,
              "padding": 8,
            },
            false,
          ]
        }
      >
        <Text
          style={
            [
              {
                "fontFamily": "MulishSemiBold",
                "fontSize": 16,
                "textAlign": "center",
              },
              false,
              false,
              [
                {
                  "fontSize": 12,
                },
                {
                  "color": "white",
                  "fontFamily": "MulishSemiBold",
                },
              ],
            ]
          }
        >
          Repay
        </Text>
      </ViewManagerAdapter_ExpoLinearGradient>
    </View>
  </View>
  <View
    style={
      [
        {
          "backgroundColor": "#CED6E1",
          "height": 1,
          "width": "100%",
        },
        undefined,
      ]
    }
  />
  <View
    style={
      {
        "flexWrap": "wrap",
        "gap": 16,
      }
    }
  >
    <View
      style={
        {
          "flexDirection": "row",
        }
      }
    >
      <View
        style={
          {
            "width": "50%",
          }
        }
      >
        <Text
          style={
            {
              "color": "#4C4C60",
              "fontFamily": "MulishRegular",
              "fontSize": 12,
            }
          }
        >
          Rate
        </Text>
        <Text
          style={
            {
              "color": "#1B1D4D",
              "fontFamily": "MulishSemiBold",
              "fontSize": 14,
            }
          }
        >
          13
        </Text>
      </View>
      <View
        style={
          {
            "width": "50%",
          }
        }
      >
        <Text
          style={
            {
              "color": "#4C4C60",
              "fontFamily": "MulishRegular",
              "fontSize": 12,
            }
          }
        >
          Period
        </Text>
        <Text
          style={
            {
              "color": "#1B1D4D",
              "fontFamily": "MulishSemiBold",
              "fontSize": 14,
            }
          }
        >
          1
           month
        </Text>
      </View>
    </View>
    <View
      style={
        {
          "flexDirection": "row",
        }
      }
    >
      <View
        style={
          {
            "width": "50%",
          }
        }
      >
        <Text
          style={
            {
              "color": "#4C4C60",
              "fontFamily": "MulishRegular",
              "fontSize": 12,
            }
          }
        >
          Monthly payment
        </Text>
        <Text
          style={
            {
              "color": "#1B1D4D",
              "fontFamily": "MulishSemiBold",
              "fontSize": 14,
            }
          }
        >
          1 117.00 USD
        </Text>
      </View>
      <View
        style={
          {
            "width": "50%",
          }
        }
      >
        <Text
          style={
            {
              "color": "#4C4C60",
              "fontFamily": "MulishRegular",
              "fontSize": 12,
            }
          }
        >
          Total paid
        </Text>
        <Text
          style={
            {
              "color": "#1B1D4D",
              "fontFamily": "MulishSemiBold",
              "fontSize": 14,
            }
          }
        >
          4 468.00 USD
        </Text>
      </View>
    </View>
  </View>
</View>
`;
